universe/Conditions.cpp

```diff
@@ -2216,7 +2221,13 @@ HasSpecial::HasSpecial(std::unique_ptr<ValueRef::ValueRef<std::string>>&& name,
     m_name(std::move(name)),
     m_since_turn_low(std::move(since_turn_low)),
     m_since_turn_high(std::move(since_turn_high))
-{}
+{
+    auto operands = {m_since_turn_low.get(), m_since_turn_high.get()};
+    SetConditionVariance(operands, m_root_candidate_invariant, m_target_invariant, m_source_invariant);
+    m_root_candidate_invariant = m_root_candidate_invariant || (!m_name || m_name->RootCandidateInvariant());
```

Comment:
Could perhaps add a non-templated `ValueRef` base class that has the `*Invariant` functions so that all could be put into the same range here?

Or even a common untemplated base class for `Condition` and `ValueRef` for the same reason?

Reply:
Sounds reasonable, but I would say this is out of scope for this PR.

Reply:
The ValueRefs need to be updated before that too.